pkg/clustermesh/types/addressing.go

```diff
@@ -218,3 +218,22 @@ func (ac AddrCluster) As20() (ac20 [20]byte) {
 func (ac AddrCluster) AsNetIP() net.IP {
 	return ac.addr.AsSlice()
 }
+
+type PrefixCluster struct {
+	prefix    netip.Prefix
+	clusterID uint32
+}
+
+func PrefixClusterFrom(addr netip.Addr, bits int, clusterID uint32) PrefixCluster {
```

Comment:
The naming is taken from the standard library's package (https://pkg.go.dev/inet.af/netaddr#IPPrefixFrom). And I intentionally make these functions look similar to them since it does `netip.XXXX` + `ClusterID` enhancement. The signature of the functions `XXXYYYFrom` is expected to be XXX and YYY in their convention. If we write a transformation function, the word represents the object that comes after `From` (e.g. AddrFromSlice takes a byte slice). In that sense, this is not that much ambiguous. WDYT?

Reply:
Aha, I see. Okay with explanation this looks fine to me.